src/main/ruby/truffleruby/core/enumerator.rb

```diff
@@ -496,11 +496,12 @@ def zip(*lists)
       end
     end
 
-    public :chunk
-    public :chunk_while
-    public :slice_after
-    public :slice_before
-    public :slice_when
+    # clone these methods from the superclass
+    alias_method :chunk, :chunk
+    alias_method :chunk_while, :chunk_while
+    alias_method :slice_after, :slice_after
+    alias_method :slice_before, :slice_before
+    alias_method :slice_when, :slice_when
```

Comment:
We use this trick in a few more places, notably enumerable.rb, module.rb and integer.rb.
`jt test spec/truffle/methods_spec.rb` should find out missing ones.

I think it would be useful to have a special `TrufflePrimitive.copy_method` or so for this, that would also set the "origin" for UnboundMethod correctly, to pretend the methods are completely separate. But that's definitely out of scope of this PR, the `alias_method` trick sounds fine.

Reply:
Interesting. All tests are passing on our CI with this patch. I'll see if it breaks locally.

Reply:
Strange. It is indeed failing. I'll check why our CI doesn't catch that.

Reply:
Done